pkg/strategy/xgap/strategy.go

```diff
@@ -328,29 +331,50 @@ func (s *Strategy) placeOrders(ctx context.Context) {
 	balances := s.tradingSession.GetAccount().Balances()
 	// try to use the bid/ask price from the trading book
 	if hasPrice {
-		// try to place orders that can adjust the position
+		// try to place orders that can adjust the position if possible
 		if s.Position != nil {
-			positionSnapshot := getPositionSnapshot(s.Position)
-			s.logger.Infof("position snapshot: %+v", positionSnapshot)
-			adjPosIsPossible, adjPosOrder := buildAdjustPositionOrder(
-				s.Symbol,
-				positionSnapshot,
-				bestBid,
-				bestAsk,
-			)
-			if balanceOk, balanceErr := tradingutil.HasSufficientBalance(
-				s.tradingMarket,
-				adjPosOrder,
-				balances,
-			); adjPosIsPossible && balanceOk {
+			if orderPlaced := func() bool {
+				positionSnapshot := getPositionSnapshot(s.Position)
+				s.logger.Infof("position snapshot: %+v", positionSnapshot)
+				adjPosIsPossible, adjPosOrder := buildAdjustPositionOrder(
+					s.Symbol,
+					positionSnapshot,
+					bestBid,
+					bestAsk,
+				)
+				if !adjPosIsPossible {
+					return false
+				}
+				// sanity checks on the order
+				// check dust quantity
+				if s.tradingMarket.IsDustQuantity(adjPosOrder.Quantity, adjPosOrder.Price) {
+					s.logger.Warnf("adjust position order is of dust quantity: %+v", adjPosOrder)
+					return false
+				}
+				// check balance
+				balanceOk, balanceErr := tradingutil.HasSufficientBalance(
+					s.tradingMarket,
+					adjPosOrder,
+					balances,
+				)
+				if !balanceOk {
+					s.logger.WithError(balanceErr).Warnf("cannot place position adjusting order: %+v", adjPosOrder)
+					return false
+				}
+
 				s.logger.Infof("adjust position order forms: %+v", adjPosOrder)
 				createdOrders, err := s.OrderExecutor.SubmitOrders(ctx, adjPosOrder)
 				if err != nil {
-					log.WithError(err).Errorf("order submit error for adjust position, created orders: %+v", createdOrders)
+					s.logger.WithError(err).Errorf("order submit error for adjust position, created orders: %+v", createdOrders)
+					if err := s.OrderExecutor.GracefulCancel(ctx, createdOrders...); err != nil {
```

Comment:
if you got error, you don't have to cancel it.

Reply:
Got it.
I forget that the order is an IOC order.